<?xml version="1.0"?>
<launch>

  <arg name="robot_id" default="1"/>
  <arg name="robot_name" default="black"/>

  <node pkg="mocap_optitrack"
    type="mocap_node"
    name="mocap_node_$(arg robot_name)"
    respawn="false"
    launch-prefix="">
    <rosparam subst_value="true">                                                                    
      rigid_bodies:                                                                                  
         '$(arg robot_id)':                                                                          
               pose: mocap/pose                                                                      
               pose2d: mocap/ground_pose                                                             
               child_frame_id: body                                                                  
               parent_frame_id: world                                                                
      optitrack_config:                                                                              
         enable_optitrack: true                                                                      
         multicast_address: 239.255.42.99                                                            
         command_port: 1510                                                                          
         data_port: 1511                                                                             
    </rosparam>
    <remap from="~/mocap/pose" to="$(arg robot_name)/pose"/>
    <remap from="~/mocap/ground_pose" to="$(arg robot_name)/ground_pose"/>
  </node>
</launch>